test/integration/scheduler/priorities_test.go

```diff
@@ -172,3 +173,66 @@ func TestPodAffinity(t *testing.T) {
 	}
 	t.Errorf("Pod %v got scheduled on an unexpected node: %v.", podName, pod.Spec.NodeName)
 }
+
+// TestImageLocality verifies that the scheduler's image locality priority function
+// works correctly, i.e., the pod gets scheduled to the node where its container images are ready.
+func TestImageLocality(t *testing.T) {
+	context := initTest(t, "image-locality")
+	defer cleanupTest(t, context)
+
+	// Add a few nodes.
+	_, err := createNodes(context.clientSet, "testnode", nil, 10)
+	if err != nil {
+		t.Fatalf("cannot create nodes: %v", err)
+	}
+
+	// We use a fake large image as the test image used by the pod, which has relatively large image size.
+	// TODO: replace this image with a LargeImage dedicated to testing image locality once the LargeImage is added
+	// TODO: change the image name to the CRI compliant name
```

Comment:
:) I will just remove this TODO as it can be misleading - the name used here is fake-large-image:v1 which might be expanded to say docker.io/fake-large-image:v1 - but since the current matching works for both cases, I use the former for simplicity.